streams/src/test/java/org/apache/kafka/test/MockInternalProcessorContext.java

```diff
@@ -105,4 +203,57 @@ public void register(final StateStore store, final StateRestoreCallback stateRes
     public StateRestoreCallback stateRestoreCallback(final String storeName) {
         return restoreCallbacks.get(storeName);
     }
+
+    public StateRestoreListener getRestoreListener(final String storeName) {
+        final StateRestoreCallback restoreCallback = restoreCallbacks.get(storeName);
+        if (restoreCallback instanceof StateRestoreListener) {
+            return (StateRestoreListener) restoreCallback;
+        }
+        return CompositeRestoreListener.NO_OP_STATE_RESTORE_LISTENER;
+    }
+
+    public void restore(final String storeName, final Iterable<KeyValue<byte[], byte[]>> changeLog) {
+        final RecordBatchingStateRestoreCallback restoreCallback = adapt(restoreCallbacks.get(storeName));
+        final StateRestoreListener restoreListener = getRestoreListener(storeName);
+
+        restoreListener.onRestoreStart(null, storeName, DEFAULT_OFFSET, DEFAULT_OFFSET);
+
+        final List<ConsumerRecord<byte[], byte[]>> records = new ArrayList<>();
+        for (final KeyValue<byte[], byte[]> keyValue : changeLog) {
+            records.add(new ConsumerRecord<>(DEFAULT_TOPIC, DEFAULT_PARTITION, DEFAULT_OFFSET, keyValue.key, keyValue.value));
+        }
+
+        restoreCallback.restoreBatch(records);
+
+        restoreListener.onRestoreEnd(null, storeName, 0L);
+    }
+
+    public void setValueSerde(final Serde valueSerde) {
+        this.valueSerde = valueSerde;
+    }
+
+    private static File createStateDir(final Properties config) {
+        if (config.containsKey(StreamsConfig.STATE_DIR_CONFIG)) {
+            return new File(config.getProperty(StreamsConfig.STATE_DIR_CONFIG));
+        }
+        return TestUtils.tempDirectory();
+    }
+
+    private static ThreadCache createThreadCache(final Properties config,
+                                                 final LogContext logContext,
+                                                 final long maxCacheSizeBytes,
+                                                 final Metrics metrics) {
+        return new ThreadCache(logContext, maxCacheSizeBytes, new StreamsMetricsImpl(metrics, DEFAULT_CLIENT_ID, getMetricsVersion(config)));
+    }
+
+    private static String getMetricsVersion(final Properties config) {
+        if (config.containsKey(StreamsConfig.BUILT_IN_METRICS_VERSION_CONFIG)) {
+            return config.getProperty(StreamsConfig.BUILT_IN_METRICS_VERSION_CONFIG);
+        }
+        return DEFAULT_METRICS_VERSION;
```

Comment:
req: Please use the following if you still need this method.
```suggestion
        return config.getProperty(StreamsConfig.BUILT_IN_METRICS_VERSION_CONFIG, DEFAULT_METRICS_VERSION);
```